reconcile/openshift_resources_base.py

```diff
@@ -308,6 +314,8 @@ def lookup_secret(
             secret_reader = SecretReader(settings)
         return secret_reader.read(secret)
     except Exception as e:
```

Comment:
Please instead catch SecretNotFound before catch generic exception and then do the if match/raise forward thing.

Reply:
Indeed. For example if the connection to vault fails for whatever reason, you do not want to return a value, but you need to raise the exception.